src/Microsoft.AspNet.TestHost/HttpResponseMessageExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.TestHost
+{
+    public static class HttpResponseMessageExtensions
+    {
+        public static Task PipelineCompleteAsync(this HttpResponseMessage message)
```

Comment:
I agree this is not necessary for implementing RequestAborted.
The test I was planning to write using RequestAborted has an effect on subsequent requests. I noticed the effects weren't visible immediately because when the ResponseStream returns, the pipeline hadn't finished yet.
I think it is an interesting feature independent of RequestAborted.
If you like, I can leave it out here and open another feature request?